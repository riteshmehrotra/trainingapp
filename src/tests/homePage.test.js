import React from 'react';
import HomePage from '../homePage';
import { render, fireEvent, waitForDomChange } from '@testing-library/react';
import renderer from 'react-test-renderer';

jest.mock('../topicProvider');


describe("Home page with language selection bar and welcome note", () => {

    test("Test that home page displayed with correct welcome note", async () => {
        let { getByText } = render(<HomePage user="David"></HomePage>);
        await waitForDomChange(() => getByText("Hello! David"));

        expect(getByText("Hello! David")).toBeInTheDocument();
    });


    test("Test that home page displayed with language bar, welcome note and topics", () => {
        const tree = renderer.create(<HomePage user="David"></HomePage>);
        const json = tree.toJSON();
        expect(json).toMatchSnapshot();
    });


    test("Test that welcome note changes on change of language preference", async () => {
        let { getByText } = render(<HomePage user="Simon"></HomePage>);
        await waitForDomChange(() => getByText("Hello! Simon"));
        expect(getByText("Hello! Simon")).toBeInTheDocument();
        fireEvent.click(getByText("ES"));

        expect(getByText("Hola! Simon")).toBeInTheDocument();
    })



    test("Test that topics are listed on home page", async () => {
        let { queryByText, getByText } = render(<HomePage user="Simon"></HomePage>);
        await waitForDomChange(() => getByText("Fake 2"));

        expect(queryByText("Fake 3")).not.toBeInTheDocument();
        expect(getByText("Fake 1")).toBeInTheDocument();
        expect(getByText("Fake 2")).toBeInTheDocument();

    })


})