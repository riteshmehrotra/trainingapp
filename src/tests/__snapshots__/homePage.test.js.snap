// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page with language selection bar and welcome note Test that home page displayed with language bar, welcome note and topics 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="ml-auto nav"
      onKeyDown={[Function]}
    >
      <h5>
        <span
          className="badge badge-primary"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          EN
        </span>
        <span
          className="badge badge-secondary"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          ES
        </span>
      </h5>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          <div>
            <h3>
              Hello
              ! 
              David
            </h3>
            <h4>
              Welcome to TDD with React training
            </h4>
          </div>
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-deck"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
